### DISCUSSION TITLE
RM520N-GLAA can't upgrade firmware with Qflash

### DISCUSSION THREAD

**Discussion by Zaedon33** [iamromulan/cellular-modem-wiki#14] [Help - Quectel] [answered]
Good afternoon,
I am attempting to set up a modem using the following components.
RM520N-GLAA module 
Board: M.2 Key B 4G 5G Modem Adapter Enclosure supports PoE power, 2.1mm barrel plug power, as well as USB port power (USB3 Micro B connector). It is the newest version of our USB3 to M.2 adapter; V8 https://store.thewirelesshaven.com/products/rj45-usb3-poe-m2-modem-adapter-v8?fbclid=IwZXh0bgNhZW0CMTEAAR3oESwzAro3aIZy0JJG4RPxCvWLM-OVCe0dd4b1peQqU5yddO_Xx-WQtbo_aem_aq4DAMRkiWmRUxnufgGx9w
Windows 11

I have it functioning as a modem. In fact I am using it right now as I type this message.
However, when I try to update the firmware It gets stuck at downloading firehose and times out at 110 secs every time. I am using Qflash!

I also went through all the ADB setup as well. Unfortunately, It doesn't seem to be working either as when running the adb devices command no devices are listed.

Any assistance would be greatly appreciated.

**Comment by Zaedon33** [comment]
I got the firmware to update. It appears that this module, like many things, doesn't play well with Windows 11. Still trying to get ADB to find the device.

**Reply by annualstatement** [reply, marked as answer]
Hi, i'm wondering if you could explain exactly how you did it. I'm having a the exact issue and its driving me nuts. I used to be able to update using my laptop with a m2 to USB adapter but that is no longer working. Did you do this on a desktop or laptop?

**Reply by Zaedon33** [reply]
I'm sorry! I haven't been on here in a while. I used a laptop with windows 10. Then just followed the instructions on this site. I used the board and case listed in my original post. I haven't had time to really mess with adb. However, I suspect it is due to the installer not creating an evironment variable path. So, the path will likely have to be entered manually to get adb to work.

**Reply by Zaedon33** [reply]
After much trying and failing I got adb to work. The problem was the installing the RNDIS driver. When the RNDIS driver installed on either windows 10 or 11 it breaks adb. The device is no longer found. Once the RNDIS driver is uninstalled and the ECM driver is re selected manually adb will work. So, to get the module up and running and adb functioning properly.

Install the NDIS, ECM and MBIM drivers. Select the ECM driver manually in Device manager.

**Reply by annualstatement** [reply]
appreciate the update, thank you!
